platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -273,7 +280,9 @@ static void mbed_minimal_formatted_string_double(char *buffer, size_t length, in
     /* get decimal part */
     double precision = 1.0;
 
-    for (size_t index = 0; index < MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS; index++) {
+    dec_precision = (dec_precision == PRECISION_DEFAULT) ? MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS : dec_precision;
```

Comment:
```suggestion
    if (dec_precision == PRECISION_DEFAULT) {
        dec_precision = MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS;
    }
```
Is this more readable?

Reply:
yes you are right